mesonbuild/build.py

```diff
@@ -119,6 +119,7 @@
 known_shmod_kwargs = known_build_target_kwargs | {'vs_module_defs'}
 known_stlib_kwargs = known_build_target_kwargs | {'pic', 'prelink'}
 known_jar_kwargs = known_exe_kwargs | {'main_class', 'java_resources'}
+known_shader_kwargs = known_build_target_kwargs
```

Comment:
Afaict this basically handles none of the build_target kwargs (as in, they are silently ignored). It shouldn't list any compatibility.

Reply:
It does use install_* and <lang>_args and it could use a few more. If shader() was separate from build_target() it would have to duplicate those.

Reply:
The point I'm trying to make is that it semantically doesn't make sense to define known_shader_kwargs in terms of build_target_kwargs, but rather in terms of what it actually accepts.

The known kwargs are just a set of kwargs that are permitted and passed along to the function rather than erroring out. We *should* be erroring out for any kwarg that the *implementation* of shader() doesn't handle.

This has nothing to do with whether shader() is separate from build_target().